packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts

```diff
@@ -455,6 +456,15 @@ export default class PoolDataAdapter {
     }
 
     try {
+      let tokenSwapAmount = totalPrincipals;
+      if (totalYields.gt(totalPrincipals)) {
+        tokenSwapAmount = totalYields;
+      }
+
+      const estimatedYields = await this.getExpectedReturnForShareToken(tempusAMM, tokenSwapAmount, true);
```

Comment:
Just to document my suggestion from call 
```suggestion
      let tokenInAmount:BigNumber;
      let yieldsIn:bool;
      if (totalYields.gt(totalPrincipals)) {
        tokenInAmount = totalYields - totalPrincipals;
        yieldsIn = true;
      } else {
        tokenInAmount = totalPrincipals - totalYields;
        yieldsIn = false;
      }

      const estimatedTokensOut = await this.getExpectedReturnForShareToken(tempusAMM, tokenSwapAmount, yieldsIn);
```